<!DOCTYPE html>



  


<html class="theme-next pisces use-motion" lang="zh-Hans">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>
<meta name="theme-color" content="#222">









<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />















  
  
  <link href="https:/AnJingwd.github.io/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />




  
  
  
  

  
    
    
  

  

  

  

  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="https:/AnJingwd.github.io/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="https:/AnJingwd.github.io/css/main.css?v=5.1.2" rel="stylesheet" type="text/css" />


  <meta name="keywords" content="NGS," />





  <link rel="alternate" href="https:/AnJingwd.github.io/atom.xml" title="安静-不安静的博客" type="application/atom+xml" />




  <link rel="shortcut icon" type="image/x-icon" href="https:/AnJingwd.github.io/favicon.ico?v=5.1.2" />






<meta name="description" content="ANNOVAR进行突变注释">
<meta name="keywords" content="NGS">
<meta property="og:type" content="article">
<meta property="og:title" content="ANNOVAR进行突变注释">
<meta property="og:url" content="https:///AnJingwd.github.io/2018/01/20/ANNOVAR进行突变注释/index.html">
<meta property="og:site_name" content="安静-不安静的博客">
<meta property="og:description" content="ANNOVAR进行突变注释">
<meta property="og:locale" content="zh-Hans">
<meta property="og:image" content="https://i.imgur.com/UGMNkMZ.jpg">
<meta property="og:image" content="https://i.imgur.com/mjzONDT.jpg">
<meta property="og:image" content="https://i.imgur.com/rISPcMC.jpg">
<meta property="og:image" content="https://i.imgur.com/vzD3s1M.jpg">
<meta property="og:image" content="https://i.imgur.com/NTrEEp8.png">
<meta property="og:updated_time" content="2018-03-06T12:43:23.643Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="ANNOVAR进行突变注释">
<meta name="twitter:description" content="ANNOVAR进行突变注释">
<meta name="twitter:image" content="https://i.imgur.com/UGMNkMZ.jpg">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: 'https:///AnJingwd.github.io/',
    scheme: 'Pisces',
    sidebar: {"position":"left","display":"post","offset":12,"offset_float":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>







  <title>ANNOVAR进行突变注释 | 安静-不安静的博客</title>
  














</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="https:///AnJingwd.github.io/"  class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">安静-不安静的博客</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <h1 class="site-subtitle" itemprop="description"></h1>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="https:/AnJingwd.github.io/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="https:/AnJingwd.github.io/categories" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br />
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="https:/AnJingwd.github.io/archives" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br />
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="https:/AnJingwd.github.io/tags" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br />
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="https:/AnJingwd.github.io/about" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br />
            
            关于
          </a>
        </li>
      
        
        <li class="menu-item menu-item-sitemap">
          <a href="https:/AnJingwd.github.io/sitemap.xml" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-sitemap"></i> <br />
            
            站点地图
          </a>
        </li>
      

      
    </ul>
  

  
</nav>




 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="https:///AnJingwd.github.iohttps:/AnJingwd.github.io/2018/01/20/ANNOVAR进行突变注释/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="AJ's blog">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="https:/AnJingwd.github.io/images/avatar.png">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="安静-不安静的博客">
    </span>

    
      <header class="post-header">

        
        
          <h2 class="post-title" itemprop="name headline">ANNOVAR进行突变注释</h2>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2018-01-20T21:50:21+08:00">
                2018-01-20
              </time>
            

            

            
          </span>

          
            <span class="post-category" >
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="https:/AnJingwd.github.io/categories/NGS/" itemprop="url" rel="index">
                    <span itemprop="name">NGS</span>
                  </a>
                </span>

                
                
              
            </span>
          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <blockquote class="blockquote-center">ANNOVAR进行突变注释</blockquote>

<a id="more"></a>
<h1 id="ANNOVAR简介"><a href="#ANNOVAR简介" class="headerlink" title="ANNOVAR简介"></a>ANNOVAR简介</h1><p><a href="http://annovar.openbioinformatics.org/en/latest/" target="_blank" rel="external">ANNOVAR Documentation</a>  &emsp;&emsp; ANNOVAR其网页版<a href="http://wannovar.wglab.org/" target="_blank" rel="external">wANNOVAR</a></p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4718734/bin/NIHMS751006-supplement-Supplementary_Text_and_Figures.pdf" target="_blank" rel="external">Comparison between ANNOVAR and other command line tools for variant annotation</a></p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4718734/bin/NIHMS751006-supplement-Supplementary_Text_and_Figures.pdf" target="_blank" rel="external">Comparison between ANNOVAR and other command line tools for variant annotation</a></p>
<p>&emsp;&emsp;例如对于人的外显子组测序，可以获得4百万个SNVs(单核苷酸突变)和50万个indels(insertion or deletions)突变。对于基因间区的突变，我们对其侧翼基因感兴趣，并想知道两者之间距离；对于外显子区域的变异，我们关系其对于氨基酸的改变。</p>
<p>&emsp;&emsp;ANNOVAR是一个perl编写的命令行工具，能对<strong>根据hg18,hg19,hg38不同版本基因组call 出来的遗传变异</strong>进行功能注释。允许多种输入文件格式，包括最常被使用的VCF格式，只需要给出chromosome, start position, end position, reference nucleotide and observed nucleotides这6个字段。输出文件也有多种格式，包括注释过的VCF文件、用tab或者逗号分隔的text文件。</p>
<p>&emsp;&emsp;ANNOVAR支持三种不同形式的注释： gene-based, region-based 和filter-based. 这三种注释分别针对于每一个variant的不同方面：</p>
<p><img src="https://i.imgur.com/UGMNkMZ.jpg" alt=""></p>
<ul>
<li>基于基因的注释（gene-based annotation）：SNPs或者CNV变异是否造成编码蛋白氨基酸的改变或者影响；</li>
<li>基于区域的注释（region-based annotation）揭示variant 与不同基因组特定区域的关系，例如：它是否与以下区域有overlap: conserved genomic elements，cytogenetic bands, microRNA target sites和Encyclopedia of DNA Elements (ENCODE)-annotated regions等</li>
<li>基于过滤的注释（ filter-based annotation ）则给出这个variant的一系列信息，例如是否该变异在dbSNP数据库报道过，该变异在1000genome项目，外显子组项目中的频率等，以及不同的计算机算法对变异注释的结果（SIFT/PolyPhen/LRT/MutationTaster/MutationAssessor/FATHMM/MetaSVM/MetaLR scores）等</li>
</ul>
<h1 id="下载（无需安转，但需要安装perl环境）"><a href="#下载（无需安转，但需要安装perl环境）" class="headerlink" title="下载（无需安转，但需要安装perl环境）"></a>下载（无需安转，但需要安装perl环境）</h1><p>最新版官网：<a href="http://annovar.openbioinformatics.org/en/latest/" target="_blank" rel="external">网址</a></p>
<pre><code>untargz annovar.latest.tar.gz
echo &apos;export PATH=/home/wangdong/softwares/annovar/annovar_latest&apos;&gt;&gt;~/.bashrc
</code></pre><p>使用ANNOVAR进行突变注释时，gene definition files(例如GTF文件) 和transcript FASTA files（FASTA文件）版本需要同步对应，否则可能会出现mis-annotated variants (such as annotating SNVs as<br>indels)</p>
<h1 id="常用注释数据库下载"><a href="#常用注释数据库下载" class="headerlink" title="常用注释数据库下载"></a>常用注释数据库下载</h1><ol>
<li><p>注释文件在ANNOVER分For gene-based annotation和For filter-based annotation以列表的形式详细介绍，<a href="http://annovar.openbioinformatics.org/en/latest/user-guide/download/" target="_blank" rel="external">链接</a></p>
<p> perl annotate_variation.pl –downdb –webfrom annovar –buildver hg19 refGene humandb/<br> perl annotate_variation.pl –downdb –webfrom annovar –buildver hg19 1000g2015aug humandb/<br> perl annotate_variation.pl –downdb –webfrom annovar –buildver hg19 exac03 humandb/<br> perl annotate_variation.pl –downdb –webfrom annovar –buildver hg19 clinvar_20170905 humandb/<br> perl annotate_variation.pl –downdb –webfrom annovar –buildver hg19 avsnp150 humandb/<br> perl annotate_variation.pl –downdb –webfrom annovar –buildver hg19 cosmic70 humandb/<br> perl annotate_variation.pl –downdb –webfrom annovar –buildver hg19 icgc21 humandb/<br> perl annotate_variation.pl –downdb –webfrom annovar –buildver hg19 nci60 humandb/</p>
</li>
</ol>
<p>这里下载的是几个通常用到的数据库的最新版：</p>
<table>
<thead>
<tr>
<th>Build</th>
<th>Table  Name</th>
<th>Explanation</th>
<th>Date</th>
</tr>
</thead>
<tbody>
<tr>
<td>hg19</td>
<td>refGene</td>
<td>FASTA sequences for all annotated transcripts in RefSeq Gene</td>
<td>20170601</td>
</tr>
<tr>
<td>hg19</td>
<td>1000g2015aug (6 data sets)</td>
<td>The 1000G team fixed a bug in chrX frequency calculation. Based on 201508 collection v5b (based on 201305 alignment)</td>
<td>20150824</td>
</tr>
<tr>
<td>hg19</td>
<td>exac03</td>
<td>ExAC 65000 exome allele frequency data for ALL, AFR (African), AMR (Admixed American), EAS (East Asian), FIN (Finnish), NFE (Non-finnish European), OTH (other), SAS (South Asian)). version 0.3. Left normalization done.</td>
<td>20151129</td>
</tr>
<tr>
<td>hg19</td>
<td>clinvar_20170905</td>
<td>Clinvar version 20170905 with separate columns (CLINSIG CLNDBN CLNACC CLNDSDB CLNDSDBID)</td>
<td>20171003</td>
</tr>
<tr>
<td>hg19</td>
<td>avsnp150</td>
<td>dbSNP150 with allelic splitting and left-normalization</td>
<td>20170929</td>
</tr>
<tr>
<td>hg19</td>
<td>cosmic70</td>
<td>COSMIC database version 70</td>
<td>20140911</td>
</tr>
<tr>
<td>hg19</td>
<td>icgc21</td>
<td>International Cancer Genome Consortium version 21</td>
<td>20160622</td>
</tr>
<tr>
<td>hg19</td>
<td>nci60</td>
<td>NCI-60 human tumor cell line panel exome sequencing allele frequency data</td>
<td>20130724</td>
</tr>
</tbody>
</table>
<ul>
<li>‘1000g2015aug’(version  August 2015), 是2015年版，包含6个 data sets，提供千人基因组项目的替换等位基因频率等信息。</li>
<li>‘exac03’(version 0.3),是0.3版外显子Exome Aggregation Consortium中报道过的variants。</li>
<li>‘clinvar_20140929’ for the variants reported in the ClinVar database (version 20140929)，ClinVar数据库整合了十多个不同类型数据库、通过标准的命名法来描述疾病，同时支持科研人员将数据下载到本地中，开展更为个性化的研究。<strong>ClinVar数据库的目的在于整合这些分散的数据、将变异、临床表型、实证数据以及功能注解与分析等四个方面的信息，通过专家评审，逐步形成一个标准的、可信的、稳定的遗传变异-临床表型相关的数据库。</strong></li>
</ul>
<ol>
<li>基于保守基因组元件注释数据文件，每一个在ANNOVAR说明文档都有详细介绍，<a href="http://annovar.openbioinformatics.org/en/latest/user-guide/region/#overview" target="_blank" rel="external">链接地址</a>，这里我们也只下载常用的即可。</li>
</ol>
<pre><code>perl annotate_variation.pl --downdb --buildver hg19 cytoBand humandb/ 
perl annotate_variation.pl -build hg19 -downdb gwasCatalog humandb/
</code></pre><ul>
<li>‘cytoBand’ 是每个细胞间band（cytogenetic band）的染色体坐标信息 ,</li>
</ul>
<p>注意：</p>
<p>1、第一个命令中不包含 ‘–webfrom annovar’ 选项, 因此是从the UCSC Genome Browser annotation database下载文件的； </p>
<p>2、 ‘–buildver hg19’ 选项是针对hg19这一版的基因组的；</p>
<p>3、运行上面命令后，在 ‘humandb/’ 目录下会多几个以 ‘hg19’为前缀的文件。 </p>
<h1 id="变异注释"><a href="#变异注释" class="headerlink" title="变异注释"></a>变异注释</h1><h2 id="ANNOVER-input-file"><a href="#ANNOVER-input-file" class="headerlink" title="ANNOVER input file"></a>ANNOVER input file</h2><p>ANNOVER的输入文件要求VCF 4.0及以上，这一要求对于绝大多数call 变异的软件，如GATK何SAMtools都是满足的。<a href="http://vcftools.sourceforge.net/specs.html" target="_blank" rel="external">VCF (Variant Call Format) specifications （包括 VCF4.0，4.1，4.2，以及BCF v2.1）
</a></p>
<h2 id="对于ANNOVAR新手，最方便使用table-annovar-pl程序"><a href="#对于ANNOVAR新手，最方便使用table-annovar-pl程序" class="headerlink" title="对于ANNOVAR新手，最方便使用table_annovar.pl程序"></a>对于ANNOVAR新手，最方便使用table_annovar.pl程序</h2><pre><code>table_annovar.pl example/ex1.avinput humandb/ -buildver hg19 -out myanno -remove -protocol refGene,cytoBand,exac03,avsnp147,dbnsfp30a -operation gx,r,f,f,f -nastring . -csvout -polish -xref example/gene_xref.txt
</code></pre><h3 id="参数解释"><a href="#参数解释" class="headerlink" title="参数解释"></a>参数解释</h3><ul>
<li>-buildver hg19 ANNOVAR默认注释是根据hg18的，所以这里需要指定hg19。</li>
<li>-out myanno 输出注释结果文件myanno.hg19_multianno.txt</li>
<li>-operation g表示gene-based；-operation gx 表示 gene-based with cross-reference annotation (from -xref argument)；-operation r 表示region-based； -operation f 表示 filter-based。如果不提供xref文件就只能指定-operation g。</li>
<li>-csvout使用逗号输出，如果喜欢使用tab键输出，则只需要去除 -csvout</li>
</ul>
<h2 id="annotate-variation-pl是-ANNOVAR的核心程序"><a href="#annotate-variation-pl是-ANNOVAR的核心程序" class="headerlink" title="annotate_variation.pl是 ANNOVAR的核心程序"></a>annotate_variation.pl是 ANNOVAR的核心程序</h2><p>使用例子：</p>
<pre><code>annotate_variation.pl -geneanno -dbtyep refGene -out ex1 -buildver hg19 example/ex1.avinput humandb/
annotate_variation.pl -regionanno -dbtype cytoBand -buildver hg19 example/ex1.avinput humandb/ 
annotate_variation.pl -filter -dbtype exac03 -buildver hg19 example/ex1.avinput humandb/
</code></pre><h3 id="参数解释及技术问题"><a href="#参数解释及技术问题" class="headerlink" title="参数解释及技术问题"></a>参数解释及技术问题</h3><ul>
<li><p>–geneanno -dbtype refGene是默认的，可以省略</p>
</li>
<li><p><strong>ncRNA above refers to RNA without coding annotation. It does not mean that this is a RNA that will never be translated.</strong> For example, BC039000 is regarded as ncRNA by ANNOVAR when using UCSC Known Gene annotation, but it is regarded as a protein-coding gene by ANNOVAR when using ENSEMBL annotation</p>
</li>
<li>By default, the gene name is printed in the second column in the variant_function file. Sometimes, a user may want to see <strong>transcript name instead. The –transcript_function argument can be used to specify this behavior</strong>. Note that it is very likely that multiple transcript names will be printed in the output separated by comma, as each gene name typically corresponds to several transcript names</li>
</ul>
<h1 id="注释结果解释"><a href="#注释结果解释" class="headerlink" title="注释结果解释"></a>注释结果解释</h1><h2 id="表头含义"><a href="#表头含义" class="headerlink" title="表头含义"></a>表头含义</h2><p><img src="https://i.imgur.com/mjzONDT.jpg" alt=""></p>
<p><img src="https://i.imgur.com/rISPcMC.jpg" alt=""></p>
<h2 id="各列内容及含义"><a href="#各列内容及含义" class="headerlink" title="各列内容及含义"></a>各列内容及含义</h2><ul>
<li>Func.refGene, Gene.refGene, GeneDetail.refGene, ExonicFunc.refGene, AAChange.refGene列包含变异对基因结构的影响。其中如果Func.refGene给出注释显示该变异位于exonic/intronic/ncRNA，那么Gene.refGene列给出相应的基因名（如果是多个基因，中间用逗号分割）；如果不是，则给出该变异侧翼两个基因及与其距离。</li>
<li>ExAC* 等列表示在外显子组项目（Exome Aggregation Consortium）数据集中所有样本，包括亚人种中的等位基因频率。</li>
<li>avsnp138 列该变异在dbSNP（version 138）数据集中的SNP 的ID号</li>
<li>其他列包含使用多个变异功能预测软件得到的预测分数，包括SIFT scores, PolyPhen2 HDIV scores, PolyPhen2 HVAR scores, LRT scores, MutationTaster scores, MutationAssessor score, FATHMM scores, GERP++ scores, CADD scores, DANN scores, PhyloP scores and SiPhy scores等。</li>
</ul>
<h2 id="Func-refGene列具体含义"><a href="#Func-refGene列具体含义" class="headerlink" title="Func.refGene列具体含义"></a>Func.refGene列具体含义</h2><table>
<thead>
<tr>
<th>Value</th>
<th>Default precedence</th>
<th>Explanation</th>
<th>Sequence Ontology</th>
</tr>
</thead>
<tbody>
<tr>
<td>exonic</td>
<td>1</td>
<td>variant overlaps a coding</td>
<td>exon_variant (SO:0001791)</td>
</tr>
<tr>
<td>splicing</td>
<td>1</td>
<td>variant is within 2-bp of a splicing junction (use -splicing_threshold to change this)</td>
<td>splicing_variant (SO:0001568)</td>
</tr>
<tr>
<td>ncRNA</td>
<td>2</td>
<td>variant overlaps a transcript without coding annotation in the gene definition (see Notes below for more explanation)</td>
<td>non_coding_transcript_variant (SO:0001619)</td>
</tr>
<tr>
<td>UTR5</td>
<td>3</td>
<td>variant overlaps a 5’ untranslated region</td>
<td>5_prime_UTR_variant (SO:0001623)</td>
</tr>
<tr>
<td>UTR3</td>
<td>3</td>
<td>variant overlaps a 3’ untranslated region</td>
<td>3_prime_UTR_variant (SO:0001624)</td>
</tr>
<tr>
<td>intronic</td>
<td>4</td>
<td>variant overlaps an intron</td>
<td>intron_variant (SO:0001627)</td>
</tr>
<tr>
<td>upstream</td>
<td>5</td>
<td><strong>variant overlaps 1-kb region upstream of transcription start site</strong></td>
<td>upstream_gene_variant (SO:0001631)</td>
</tr>
<tr>
<td>downstream</td>
<td>5</td>
<td><strong>variant overlaps 1-kb region downtream of transcription end site (use -neargene to change this)</strong></td>
<td>downstream_gene_variant (SO:0001632)</td>
</tr>
<tr>
<td>intergenic</td>
<td>6</td>
<td>variant is in intergenic region</td>
<td>intergenic_variant (SO:0001628)</td>
</tr>
</tbody>
</table>
<p><strong>The precedence defined above is used to decide what function to print out when a variant fit multiple functional categories</strong></p>
<h2 id="ExonicFunc-refGene列具体含义"><a href="#ExonicFunc-refGene列具体含义" class="headerlink" title="ExonicFunc.refGene列具体含义"></a>ExonicFunc.refGene列具体含义</h2><table>
<thead>
<tr>
<th>Value</th>
<th>Default precedence</th>
<th>Explanation</th>
<th>Sequence Ontology</th>
</tr>
</thead>
<tbody>
<tr>
<td>frameshift insertion</td>
<td>1</td>
<td>an insertion of one or more nucleotides that cause frameshift changes in protein coding sequence</td>
<td>frameshift_elongation (SO:0001909)</td>
</tr>
<tr>
<td>frameshift deletion</td>
<td>2</td>
<td>a deletion of one or more nucleotides that cause frameshift changes in protein coding sequence</td>
<td>frameshift_truncation (SO:0001910)</td>
</tr>
<tr>
<td>frameshift block substitution</td>
<td>3</td>
<td>a block substitution of one or more nucleotides that cause frameshift changes in protein coding sequence</td>
<td>frameshift_variant (SO:0001589)</td>
</tr>
<tr>
<td>stopgain</td>
<td>4</td>
<td>a nonsynonymous SNV, frameshift insertion/deletion, nonframeshift insertion/deletion or block substitution that lead to the immediate creation of stop codon at the variant site. For frameshift mutations, the creation of stop codon downstream of the variant will not be counted as “stopgain”!</td>
<td>stop_gained (SO:0001587)</td>
</tr>
<tr>
<td>stoploss</td>
<td>5</td>
<td>a nonsynonymous SNV, frameshift insertion/deletion, nonframeshift insertion/deletion or block substitution that lead to the immediate elimination of stop codon at the variant site</td>
<td>stop_lost (SO:0001578)</td>
</tr>
<tr>
<td>nonframeshift insertion</td>
<td>6</td>
<td>an insertion of 3 or multiples of 3 nucleotides that do not cause frameshift changes in protein coding sequence</td>
<td>inframe_insertion (SO:0001821)</td>
</tr>
<tr>
<td>nonframeshift deletion</td>
<td>7</td>
<td>a deletion of 3 or mutliples of 3 nucleotides that do not cause frameshift changes in protein coding sequence</td>
<td>inframe_deletion (SO:0001822)</td>
</tr>
<tr>
<td>nonframeshift block substitution</td>
<td>8</td>
<td>a block substitution of one or more nucleotides that do not cause frameshift changes in protein</td>
<td>coding sequence    inframe_variant (SO:0001650)</td>
</tr>
<tr>
<td>nonsynonymous SNV</td>
<td>9</td>
<td>a single nucleotide change that cause an amino acid change</td>
<td>missense_variant (SO:0001583)</td>
</tr>
<tr>
<td>synonymous SNV</td>
<td>10</td>
<td>a single nucleotide change that does not cause an amino acid change</td>
<td>synonymous_variant (SO:0001819)</td>
</tr>
<tr>
<td>unknown</td>
<td>11</td>
<td>unknown function (due to various errors in the gene structure definition in the database file)</td>
<td>sequence_variant (SO:0001060)</td>
</tr>
</tbody>
</table>
<h2 id="A-unified-annotation-format"><a href="#A-unified-annotation-format" class="headerlink" title="A unified annotation format"></a>A unified annotation format</h2><p>不同的注释软件对突变注释的表示方式不统一，不利于交流，为此，多个注释软件的作者协定了一份统一的注释格式，<a href="http://snpeff.sourceforge.net/VCFannotationformat_v1.0.pdf" target="_blank" rel="external">文档链接</a></p>
<h1 id="突变有害性注释结果选择建议（ANNOVER作者）"><a href="#突变有害性注释结果选择建议（ANNOVER作者）" class="headerlink" title="突变有害性注释结果选择建议（ANNOVER作者）"></a>突变有害性注释结果选择建议（ANNOVER作者）</h1><ul>
<li>一般而言，建议使用one popular prediction score (such as SIFT) 和 one meta-score (such as metaSVM)预测来判断突变是否是有害的</li>
<li>对于 noncoding variants的有害性预测推荐使用one popular prediction score (such as PhyloP) 和 one meta-score (such as CADD)</li>
</ul>
<h1 id="补充"><a href="#补充" class="headerlink" title="补充"></a>补充</h1><p>vcf文件经过Annovar注释后突变个数应该相同，但对于原vcf文件类似如下情况，Annovar则会分别注释成两行</p>
<p><img src="https://i.imgur.com/vzD3s1M.jpg" alt=""></p>
<p>在Annovar官网的FAQ已经给出解释：</p>
<p><img src="https://i.imgur.com/NTrEEp8.png" alt=""></p>
<p>这里需要重新了解下vcf文件格式，（<a href="http://annovar.openbioinformatics.org/en/latest/articles/VCF/" target="_blank" rel="external">Annovar VCF Processing Guide</a>）：</p>
<ol>
<li>GT：样品的基因型（genotype）。两个数字中间用’/‘分 开，这两个数字表示双倍体的sample的基因型。0 表示样品中有ref的allele； 1 表示样品中variant的allele； <strong>2表示有第二个variant的allele</strong></li>
<li>VCF文件是描述位点的格式文件。尽管称作变异检出格式，但并不是为了描述变异或者基因型检出。其可以作为包含基因型检出（甚至变异）的一种选择，并且对于很多肺二倍体物种或者喝多情况（例如线粒体或者人的肿瘤），其有时候甚至不能说是包含基因型检出。但是，很多变异检出软件的目的是为了产生基因型检出；它们使用VCF作为输出文件，但是这并不意味着VCF文件是未存储变异检出而专门设计的。</li>
<li>因为VCF是一个位点描述文件，具有很多的结果。首先，并不是一行对应着一个变异。因为多个变异能在相同的位点，VCF文件的一行原则上可以描述多个突变（包含野生型非突变allele），和多种类型的基因型检出。例如，看下面这来自VCF文件的一行、其有8个tab键分割的列。在ALT列，具有数个逗号分割的改变的alleles。因此在单行上，数个插入和删除，以及一个单核苷酸（SNV）突变同时存在。</li>
</ol>
<p>1 112240038 . CTTT CTTTT,CTTTTT,CTTA,CTT,CT,C . PASS AC=986,3,1249,3,127,3;AF=0.196885,0.000599042,0.249401,0.000599042,0.0253594,0.000599042</p>
<p>很多Annovar使用者喜欢保留VCF文件中的突变注释信息（通过INFO列）。因此对于上述这种情况，我们需要在相同行的INFO列加注释信息到所有的6个allels，并且确保使用者知道相应的注释对应相应的allele</p>
<ol>
<li>VCF可以hijack你的变异，使得SNVs变为多核苷酸突变，并且使得简单的indels变的复杂。这将对注释造成困难，举上述的 CTTT-&gt;CTTA 改变作为例子，其应该是简单的T&gt;A （SNV），但是因为deletion/insertion hijack the locus，其被写为CTTT-&gt;CTTA 而不是T-&gt;A 。考虑到单个allele频率数据库（例如千人基因组项目频率数据库）将只具有 T-&gt;A 但没有 CTTT-&gt;CTTA，然后这个变异通过注释软件注释后将被遗漏，甚至其确实在1000G中被观察到过。</li>
</ol>
<p>同样的，CTTT-&gt;CTTTT 变异可能在数据库中没有记录，因为C&gt;CT可能是更合适的记录这个变异的方式。</p>
<ol>
<li>目前为止还没有描述indel的普遍认同的特定方式。很多使用者偏好做left-normalization（变异开始的位置应该尽可能向左，直至不能再左，这样数目越少越好）。但是HGVS明确指明了left-normalization将被用在cDNA（mRNA）的坐标上,这意味着 right-normalization被要求使用在人的基因组一半的基因上。</li>
<li>在阅读了这些事实后，现在的问题时我们应该如何注释VCF文件来确保最准确的结果呢？</li>
</ol>
<p>因为left-normalization变得越来越流行，我的建议是是就使用left-normalization。我的第二个建议是各个VCF行仅描述一个变异，以便indel不能隐藏SNPs，以此确保1对1的和数据库匹配。</p>
<p>我作为ANNOVAR开发者决定去预处理所有的1000G文件，ESP6500si文件和dbSNP文件，以便一行包含一个变异，使得每个变异都采取 left-normalized。已经更新的数据再2014年的12月可以获得了。</p>
<p>作为使用者我们应该这样做：（1）分割VCF文件确保每一行只包含一个变异；（2）left-normalize所有的VCF文件； （3）使用ANNOVAR注释</p>
<p>所以使用命令：</p>
<p>bcftools norm -m-both -o ex1.step1.vcf ex1.vcf.gz</p>
<p>bcftools norm -f human_g1k_v37.fasta -o ex1.step2.vcf ex1.step1.vcf</p>
<p>第一个命令分割多allels变异检出为单独的行，第二个命令运行真正的 left-normalization。（有时候第一个命令可能出现没有变异能被分解，尽管在文件中存在这些变异，这种情况下，你可以使用<a href="http://genome.sph.umich.edu/wiki/Vt" target="_blank" rel="external">vt program</a> 代替）</p>
<ol>
<li>目前使用left-normalized的数据集包括：avsnp138，avsnp142，clinvar_20150330<br>，1000g2014oct，exac03，esp6500siv2</li>
</ol>
<h1 id="参考"><a href="#参考" class="headerlink" title="参考"></a>参考</h1><p>（1）<a href="https://www.ncbi.nlm.nih.gov/pubmed/26379229" target="_blank" rel="external">Genomic variant annotation and prioritization with ANNOVAR and wANNOVAR</a></p>
<p>（2）<a href="http://blog.sina.com.cn/s/blog_670445240101j0pa.html" target="_blank" rel="external">每日一生信–基于保守性和规则性的预测方法SIFT和PolyPhen</a></p>
<p>（3）<a href="http://blog.csdn.net/g863402758/article/details/75304391" target="_blank" rel="external">annovar对人类基因组和非人类基因组variants注释流程</a></p>
<p>（4）<a href="http://ishare.iask.sina.com.cn/f/63732396.html" target="_blank" rel="external">ANNOVAR注释上机（推荐）</a></p>

      
    </div>
    
    
    

    

    

    
	<div>
	
		<div>
    
        <div style="text-align:center;color: #ccc;font-size:14px;">-------------本文结束<i class="fa fa-paw"></i>感谢您的阅读-------------</div>
    
</div>
	
	</div>
    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="https:/AnJingwd.github.io/tags/NGS/" rel="tag"><i class="fa fa-tag"></i> NGS</a>
          
        </div>
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="https:/AnJingwd.github.io/2018/01/10/RNA-seq质控/" rel="next" title="RNA-seq质控">
                <i class="fa fa-chevron-left"></i> RNA-seq质控
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="https:/AnJingwd.github.io/2018/01/21/批量生成与提交shell脚本/" rel="prev" title="批量生成与提交shell脚本">
                批量生成与提交shell脚本 <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
        <!-- JiaThis Button BEGIN -->
<div class="jiathis_style">
  <a class="jiathis_button_tsina"></a>
  <a class="jiathis_button_tqq"></a>
  <a class="jiathis_button_weixin"></a>
  <a class="jiathis_button_cqq"></a>
  <a class="jiathis_button_douban"></a>
  <a class="jiathis_button_renren"></a>
  <a class="jiathis_button_qzone"></a>
  <a class="jiathis_button_kaixin001"></a>
  <a class="jiathis_button_copy"></a>
  <a href="http://www.jiathis.com/share" class="jiathis jiathis_txt jiathis_separator jtico jtico_jiathis" target="_blank"></a>
  <a class="jiathis_counter_style"></a>
</div>
<script type="text/javascript" >
  var jiathis_config={
    hideMore:false
  }
</script>
<script type="text/javascript" src="http://v3.jiathis.com/code/jia.js" charset="utf-8"></script>
<!-- JiaThis Button END -->

      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap" >
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image"
               src="https:/AnJingwd.github.io/images/avatar.png"
               alt="AJ's blog" />
          <p class="site-author-name" itemprop="name">AJ's blog</p>
           
              <p class="site-description motion-element" itemprop="description"></p>
           
        </div>
        <nav class="site-state motion-element">

          
            <div class="site-state-item site-state-posts">
              <a href="https:/AnJingwd.github.io/archives">
                <span class="site-state-item-count">50</span>
                <span class="site-state-item-name">日志</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-categories">
              <a href="https:/AnJingwd.github.io/categories/index.html">
                <span class="site-state-item-count">14</span>
                <span class="site-state-item-name">分类</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-tags">
              <a href="https:/AnJingwd.github.io/tags/index.html">
                <span class="site-state-item-count">14</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        
          <div class="feed-link motion-element">
            <a href="https:/AnJingwd.github.io/atom.xml" rel="alternate">
              <i class="fa fa-rss"></i>
              RSS
            </a>
			 <a title="收藏到书签，偶尔High一下^_^" rel="alternate" class="mw-harlem_shake_slow wobble shake" href='javascript:(function() {
    function c() {
        var e = document.createElement("link");
        e.setAttribute("type", "text/css");
        e.setAttribute("rel", "stylesheet");
        e.setAttribute("href", f);
        e.setAttribute("class", l);
        document.body.appendChild(e)
    }
 
    function h() {
        var e = document.getElementsByClassName(l);
        for (var t = 0; t < e.length; t++) {
            document.body.removeChild(e[t])
        }
    }
 
    function p() {
        var e = document.createElement("div");
        e.setAttribute("class", a);
        document.body.appendChild(e);
        setTimeout(function() {
            document.body.removeChild(e)
        }, 100)
    }
 
    function d(e) {
        return {
            height : e.offsetHeight,
            width : e.offsetWidth
        }
    }
 
    function v(i) {
        var s = d(i);
        return s.height > e && s.height < n && s.width > t && s.width < r
    }
 
    function m(e) {
        var t = e;
        var n = 0;
        while (!!t) {
            n += t.offsetTop;
            t = t.offsetParent
        }
        return n
    }
 
    function g() {
        var e = document.documentElement;
        if (!!window.innerWidth) {
            return window.innerHeight
        } else if (e && !isNaN(e.clientHeight)) {
            return e.clientHeight
        }
        return 0
    }
 
    function y() {
        if (window.pageYOffset) {
            return window.pageYOffset
        }
        return Math.max(document.documentElement.scrollTop, document.body.scrollTop)
    }
 
    function E(e) {
        var t = m(e);
        return t >= w && t <= b + w
    }
 
    function S() {
        var e = document.createElement("audio");
        e.setAttribute("class", l);
        e.src = i;
        e.loop = false;
        e.addEventListener("canplay", function() {
            setTimeout(function() {
                x(k)
            }, 500);
            setTimeout(function() {
                N();
                p();
                for (var e = 0; e < O.length; e++) {
                    T(O[e])
                }
            }, 15500)
        }, true);
        e.addEventListener("ended", function() {
            N();
            h()
        }, true);
        e.innerHTML = " <p>If you are reading this, it is because your browser does not support the audio element. We recommend that you get a new browser.</p> <p>";
        document.body.appendChild(e);
        e.play()
    }
 
    function x(e) {
        e.className += " " + s + " " + o
    }
 
    function T(e) {
        e.className += " " + s + " " + u[Math.floor(Math.random() * u.length)]
    }
 
    function N() {
        var e = document.getElementsByClassName(s);
        var t = new RegExp("\\b" + s + "\\b");
        for (var n = 0; n < e.length; ) {
            e[n].className = e[n].className.replace(t, "")
        }
    }
 
    var e = 30;
    var t = 30;
    var n = 350;
    var r = 350;
    var i = "//s3.amazonaws.com/moovweb-marketing/playground/harlem-shake.mp3";
    var s = "mw-harlem_shake_me";
    var o = "im_first";
    var u = ["im_drunk", "im_baked", "im_trippin", "im_blown"];
    var a = "mw-strobe_light";
    var f = "//s3.amazonaws.com/moovweb-marketing/playground/harlem-shake-style.css";
    var l = "mw_added_css";
    var b = g();
    var w = y();
    var C = document.getElementsByTagName("*");
    var k = null;
    for (var L = 0; L < C.length; L++) {
        var A = C[L];
        if (v(A)) {
            if (E(A)) {
                k = A;
                break
            }
        }
    }
    if (A === null) {
        console.warn("Could not find a node of the right size. Please try a different page.");
        return
    }
    c();
    S();
    var O = [];
    for (var L = 0; L < C.length; L++) {
        var A = C[L];
        if (v(A)) {
            O.push(A)
        }
    }
})()'><i class="fa fa-music"></i> High</a>
          </div>
        

        <div class="links-of-author motion-element">
          
        </div>

        
        

        
        
          <div class="links-of-blogroll motion-element links-of-blogroll-inline">
            <div class="links-of-blogroll-title">
              <i class="fa  fa-fw fa-globe"></i>
              推荐阅读
            </div>
            <ul class="links-of-blogroll-list">
              
                <li class="links-of-blogroll-item">
                  <a href="http://blog.sina.com.cn/u/1253988497" title="种田农民的博客" target="_blank">种田农民的博客</a>
                </li>
              
                <li class="links-of-blogroll-item">
                  <a href="https://www.biostars.org/t/Tools/" title="Biostars" target="_blank">Biostars</a>
                </li>
              
                <li class="links-of-blogroll-item">
                  <a href="https://stackoverflow.com/" title="Stack Overflow" target="_blank">Stack Overflow</a>
                </li>
              
                <li class="links-of-blogroll-item">
                  <a href="https://segmentfault.com/" title="SegmentFault" target="_blank">SegmentFault</a>
                </li>
              
                <li class="links-of-blogroll-item">
                  <a href="http://www.alloyteam.com/nav/" title="Web前端导航" target="_blank">Web前端导航</a>
                </li>
              
                <li class="links-of-blogroll-item">
                  <a href="http://www.36zhen.com/t?id=3448" title="前端书籍资料" target="_blank">前端书籍资料</a>
                </li>
              
            </ul>
          </div>
        
		
<div id="music163player">		
	<iframe frameborder="no" border="0" 
	marginwidth="0" marginheight="0" width=330 height=86 
src="//music.163.com/outchain/player?type=2&id=471682981&auto=1&height=66">
	</iframe>
</div>	
        

      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#ANNOVAR简介"><span class="nav-number">1.</span> <span class="nav-text">ANNOVAR简介</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#下载（无需安转，但需要安装perl环境）"><span class="nav-number">2.</span> <span class="nav-text">下载（无需安转，但需要安装perl环境）</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#常用注释数据库下载"><span class="nav-number">3.</span> <span class="nav-text">常用注释数据库下载</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#变异注释"><span class="nav-number">4.</span> <span class="nav-text">变异注释</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#ANNOVER-input-file"><span class="nav-number">4.1.</span> <span class="nav-text">ANNOVER input file</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#对于ANNOVAR新手，最方便使用table-annovar-pl程序"><span class="nav-number">4.2.</span> <span class="nav-text">对于ANNOVAR新手，最方便使用table_annovar.pl程序</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#参数解释"><span class="nav-number">4.2.1.</span> <span class="nav-text">参数解释</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#annotate-variation-pl是-ANNOVAR的核心程序"><span class="nav-number">4.3.</span> <span class="nav-text">annotate_variation.pl是 ANNOVAR的核心程序</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#参数解释及技术问题"><span class="nav-number">4.3.1.</span> <span class="nav-text">参数解释及技术问题</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#注释结果解释"><span class="nav-number">5.</span> <span class="nav-text">注释结果解释</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#表头含义"><span class="nav-number">5.1.</span> <span class="nav-text">表头含义</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#各列内容及含义"><span class="nav-number">5.2.</span> <span class="nav-text">各列内容及含义</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Func-refGene列具体含义"><span class="nav-number">5.3.</span> <span class="nav-text">Func.refGene列具体含义</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#ExonicFunc-refGene列具体含义"><span class="nav-number">5.4.</span> <span class="nav-text">ExonicFunc.refGene列具体含义</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#A-unified-annotation-format"><span class="nav-number">5.5.</span> <span class="nav-text">A unified annotation format</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#突变有害性注释结果选择建议（ANNOVER作者）"><span class="nav-number">6.</span> <span class="nav-text">突变有害性注释结果选择建议（ANNOVER作者）</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#补充"><span class="nav-number">7.</span> <span class="nav-text">补充</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#参考"><span class="nav-number">8.</span> <span class="nav-text">参考</span></a></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <script async src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>

<div class="copyright" >
  
  &copy;  2017 - 
  <span itemprop="copyrightYear">2018</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">AJ's blog</span>
</div>


<!--<div class="powered-by">
  由 <a class="theme-link" href="https://hexo.io">Hexo</a> 强力驱动
</div>

<div class="theme-info">
  主题 -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Pisces
  </a>
</div>
-->


<div class="theme-info">
  <div class="powered-by"></div>
  <span class="post-count">博客全站共66.3k字</span>
</div>

<div class="powered-by">
<i class="fa fa-user-md"></i><span id="busuanzi_container_site_uv">
  本站访客数:<span id="busuanzi_value_site_uv"></span>
</span>
</div>

        

        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  






  
  







  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/three/three.min.js"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/three/canvas_lines.min.js"></script>


  


  <script type="text/javascript" src="https:/AnJingwd.github.io/js/src/utils.js?v=5.1.2"></script>

  <script type="text/javascript" src="https:/AnJingwd.github.io/js/src/motion.js?v=5.1.2"></script>



  
  


  <script type="text/javascript" src="https:/AnJingwd.github.io/js/src/affix.js?v=5.1.2"></script>

  <script type="text/javascript" src="https:/AnJingwd.github.io/js/src/schemes/pisces.js?v=5.1.2"></script>



  
  <script type="text/javascript" src="https:/AnJingwd.github.io/js/src/scrollspy.js?v=5.1.2"></script>
<script type="text/javascript" src="https:/AnJingwd.github.io/js/src/post-details.js?v=5.1.2"></script>



  


  <script type="text/javascript" src="https:/AnJingwd.github.io/js/src/bootstrap.js?v=5.1.2"></script>



  


  




	





  





  






  





  

  

  

  

  

  


</body>
</html>

<!-- 页面点击小红心 -->
<script type="text/javascript" src="/js/src/love.js"></script>
