<!DOCTYPE html>



  


<html class="theme-next pisces use-motion" lang="zh-Hans">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>
<meta name="theme-color" content="#222">









<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />















  
  
  <link href="https:/AnJingwd.github.io/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />




  
  
  
  

  
    
    
  

  

  

  

  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="https:/AnJingwd.github.io/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="https:/AnJingwd.github.io/css/main.css?v=5.1.2" rel="stylesheet" type="text/css" />


  <meta name="keywords" content="NGS," />





  <link rel="alternate" href="https:/AnJingwd.github.io/atom.xml" title="安静-不安静的博客" type="application/atom+xml" />




  <link rel="shortcut icon" type="image/x-icon" href="https:/AnJingwd.github.io/favicon.ico?v=5.1.2" />






<meta name="description" content="变异检测概述">
<meta name="keywords" content="NGS">
<meta property="og:type" content="article">
<meta property="og:title" content="GATK变异检测概述">
<meta property="og:url" content="https:///AnJingwd.github.io/2017/12/26/GATK变异检测概述/index.html">
<meta property="og:site_name" content="安静-不安静的博客">
<meta property="og:description" content="变异检测概述">
<meta property="og:locale" content="zh-Hans">
<meta property="og:image" content="https://i.imgur.com/W3L0QGg.jpg">
<meta property="og:image" content="https://i.imgur.com/9fKpXfy.jpg">
<meta property="og:image" content="https://i.imgur.com/v0ZLr2G.jpg">
<meta property="og:image" content="https://i.imgur.com/iFODOKU.jpg">
<meta property="og:image" content="https://i.imgur.com/cIZbbh2.jpg">
<meta property="og:image" content="https://i.imgur.com/r3ayrha.jpg">
<meta property="og:image" content="https://i.imgur.com/oWRVjhD.jpg">
<meta property="og:image" content="https://i.imgur.com/N5kPt0j.jpg">
<meta property="og:image" content="https://i.imgur.com/Bzem6S3.png">
<meta property="og:image" content="https://i.imgur.com/murEihF.jpg">
<meta property="og:image" content="https://i.imgur.com/Zry1hvH.jpg">
<meta property="og:image" content="https://i.imgur.com/w7BsDj9.jpg">
<meta property="og:image" content="https://i.imgur.com/lsEMXQk.jpg">
<meta property="og:image" content="https://i.imgur.com/ulVOTdY.jpg">
<meta property="og:image" content="https://i.imgur.com/FkTZcp5.jpg">
<meta property="og:image" content="https://i.imgur.com/CyxQF14.jpg">
<meta property="og:image" content="https://i.imgur.com/BV54Ah7.jpg">
<meta property="og:image" content="https://i.imgur.com/lKHfD2r.jpg">
<meta property="og:image" content="https://i.imgur.com/vLF563q.jpg">
<meta property="og:image" content="https://i.imgur.com/swA19TJ.jpg">
<meta property="og:image" content="https://i.imgur.com/P8oeuDk.jpg">
<meta property="og:image" content="https://i.imgur.com/UKTedbs.png">
<meta property="og:image" content="https://i.imgur.com/LouIHI6.jpg">
<meta property="og:image" content="https://i.imgur.com/tCv4gpU.jpg">
<meta property="og:image" content="https://i.imgur.com/EoGdjYO.jpg">
<meta property="og:image" content="https://i.imgur.com/PuPNxd8.jpg">
<meta property="og:image" content="https://i.imgur.com/3Op72W9.jpg">
<meta property="og:image" content="https://i.imgur.com/tQP2gCd.jpg">
<meta property="og:image" content="https://i.imgur.com/kSzUwtQ.jpg">
<meta property="og:image" content="https://i.imgur.com/4gy1NSD.jpg">
<meta property="og:image" content="https://i.imgur.com/MjvZPu0.jpg">
<meta property="og:image" content="https://i.imgur.com/WIRa8Pq.jpg">
<meta property="og:image" content="https://i.imgur.com/p9i0C4X.jpg">
<meta property="og:image" content="https://i.imgur.com/oDZl1f7.jpg">
<meta property="og:image" content="https://i.imgur.com/p69fkwu.jpg">
<meta property="og:updated_time" content="2018-03-11T00:49:20.278Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="GATK变异检测概述">
<meta name="twitter:description" content="变异检测概述">
<meta name="twitter:image" content="https://i.imgur.com/W3L0QGg.jpg">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: 'https:///AnJingwd.github.io/',
    scheme: 'Pisces',
    sidebar: {"position":"left","display":"post","offset":12,"offset_float":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>







  <title>GATK变异检测概述 | 安静-不安静的博客</title>
  














</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="https:///AnJingwd.github.io/"  class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">安静-不安静的博客</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <h1 class="site-subtitle" itemprop="description"></h1>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="https:/AnJingwd.github.io/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="https:/AnJingwd.github.io/categories" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br />
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="https:/AnJingwd.github.io/archives" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br />
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="https:/AnJingwd.github.io/tags" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br />
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="https:/AnJingwd.github.io/about" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br />
            
            关于
          </a>
        </li>
      
        
        <li class="menu-item menu-item-sitemap">
          <a href="https:/AnJingwd.github.io/sitemap.xml" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-sitemap"></i> <br />
            
            站点地图
          </a>
        </li>
      

      
    </ul>
  

  
</nav>




 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="https:///AnJingwd.github.iohttps:/AnJingwd.github.io/2017/12/26/GATK变异检测概述/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="AJ's blog">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="https:/AnJingwd.github.io/images/avatar.png">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="安静-不安静的博客">
    </span>

    
      <header class="post-header">

        
        
          <h2 class="post-title" itemprop="name headline">GATK变异检测概述</h2>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2017-12-26T20:58:54+08:00">
                2017-12-26
              </time>
            

            

            
          </span>

          
            <span class="post-category" >
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="https:/AnJingwd.github.io/categories/NGS/" itemprop="url" rel="index">
                    <span itemprop="name">NGS</span>
                  </a>
                </span>

                
                
              
            </span>
          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <blockquote class="blockquote-center">变异检测概述</blockquote>

<a id="more"></a>
<h1 id="突变发现背景介绍"><a href="#突变发现背景介绍" class="headerlink" title="突变发现背景介绍"></a>突变发现背景介绍</h1><p>个体相对于参考基因组的两种突变</p>
<ul>
<li>Germline    (inherited)    </li>
<li>Somatic    (cancer)    </li>
</ul>
<h2 id="不同的突变类型"><a href="#不同的突变类型" class="headerlink" title="不同的突变类型"></a>不同的突变类型</h2><p><img src="https://i.imgur.com/W3L0QGg.jpg" alt=""></p>
<h2 id="实验策略"><a href="#实验策略" class="headerlink" title="实验策略"></a>实验策略</h2><ol>
<li>全基因组测序</li>
<li>全外显子测序，或者设计目标基因的panel进行目标区域捕获测序</li>
</ol>
<p><img src="https://i.imgur.com/9fKpXfy.jpg" alt=""></p>
<p>HTS    machine    processes a    flowcell containing    lanes;each    lane cons)tutes    a read group (RG) (unless multiplexed)</p>
<h2 id="Raw-sequence的质控"><a href="#Raw-sequence的质控" class="headerlink" title="Raw sequence的质控"></a>Raw sequence的质控</h2><p>目的：将ASCII code转换为Phred-scale Q    scores，考察测序质量（错误率）</p>
<p><img src="https://i.imgur.com/v0ZLr2G.jpg" alt=""></p>
<p><strong>所以phred score一般选择20即可</strong></p>
<p><img src="https://i.imgur.com/iFODOKU.jpg" alt=""></p>
<h1 id="胚系突变snp-amp-indel分析流程"><a href="#胚系突变snp-amp-indel分析流程" class="headerlink" title="胚系突变snp&amp;indel分析流程"></a>胚系突变snp&amp;indel分析流程</h1><ul>
<li>Map to reference</li>
<li>Sort, Index</li>
<li>Mark Duplicates &amp; Sort (Picard)</li>
<li>Indels Realigment</li>
<li>Base Quality Score Recalibration</li>
<li>Variant Calling (HaplotypeCaller)</li>
<li>Variant Filtration (Hard Filter, see <a href="http://gatkforums.broadinstitute.org/gatk/discussion/2806/howto-apply-hard-filters-to-a-call-set" target="_blank" rel="external">GATK Best Practices</a>)</li>
<li>Annotation</li>
</ul>
<p><img src="https://i.imgur.com/cIZbbh2.jpg" alt=""></p>
<p>pipline:<a href="https://github.com/joergen7/variant-call-gatk" target="_blank" rel="external">variant-call-gatk</a></p>
<p>文献：<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4243306/" target="_blank" rel="external">From FastQ data to high confidence variant calls: the Genome Analysis Toolkit best practices pipeline</a></p>
<h2 id="（1）Map"><a href="#（1）Map" class="headerlink" title="（1）Map"></a>（1）Map</h2><ul>
<li>BWA    for    DNA    </li>
<li>STAR    for    RNAseq</li>
</ul>
<p>Sequence/Binary    Alignment    Map    (SAM/BAM)    </p>
<p><img src="https://i.imgur.com/r3ayrha.jpg" alt=""></p>
<h3 id="细节："><a href="#细节：" class="headerlink" title="细节："></a>细节：</h3><p>All BAM/CRAM files must satisfy the following requirements:</p>
<ul>
<li>It must be aligned to one of the references described here.</li>
<li>It must be <strong>sorted in coordinate order</strong> (not by queryname and not “unsorted”).</li>
<li>It must list the read groups with sample names in the header.</li>
<li>Every read must belong to a <strong>read group</strong>.</li>
<li>The BAM file must pass Picard ValidateSamFile validation.</li>
</ul>
<p>具体参考：<a href="https://gatkforums.broadinstitute.org/gatk/discussion/1317/collected-faqs-about-bam-files" target="_blank" rel="external">Collected FAQs about input files for sequence read data (BAM/CRAM)</a></p>
<ul>
<li>Add read group 对于多样本检测变异，比如mutect是必须的，否则当场一个样品处理</li>
<li>You only need to run FixMateInformation if you are getting errors with mate information from Picard’s ValidateSamFile， <a href="https://gatkforums.broadinstitute.org/gatk/discussion/6047/necessary-to-run-fixmateinformation" target="_blank" rel="external">具体参考</a></li>
</ul>
<h2 id="（2）Mark-duplicates"><a href="#（2）Mark-duplicates" class="headerlink" title="（2）Mark    duplicates"></a>（2）Mark    duplicates</h2><p>测序原理是随机打断，那么理论上出现两条完全相同的read的概率是非常低的，而且建库时PCR扩增存在偏向性，因此标出完全相同的read。这一步是为了减少duplication artifacts,从而正确评估alleles。duplicates的状态通过SAM flag来标记，但并没有去除（绝大多数的GATK tools默认忽 duplicates）</p>
<p>duplicates来源：（1）PCR duplicates ; (2) Optical duplicates</p>
<p>以下两种情况不能 mark duplicates:</p>
<p>Amplicon    sequencing</p>
<ul>
<li>all    reads    start    at    same    posi7on    by    design    </li>
<li>UMIAwareMarkDuplicatesWithMateCigar</li>
</ul>
<p>RNAseq    allele-specific    expression    analysis    </p>
<ul>
<li>(ASEReadCounter    can    disable    DuplicateFilter)    </li>
</ul>
<p>Duplicates    =    non-independent    measurements of    a sequenc fragment</p>
<p><img src="https://i.imgur.com/oWRVjhD.jpg" alt=""></p>
<h2 id="（3）Local-realignment"><a href="#（3）Local-realignment" class="headerlink" title="（3）Local realignment"></a>（3）Local realignment</h2><p>对于indel区域进行局部的重新比对，为了纠正比对错误。</p>
<p><img src="https://i.imgur.com/N5kPt0j.jpg" alt=""></p>
<p>Mappers对于在reads末端的indel较难检出，因为错配的罚分为-1，而open gap的罚分为-3。咋局部的indel realignment后能减少假阳性的SNPs数量。</p>
<p>具体过程：</p>
<p>（1）通过RealignerTargetCreator，根据1000    Genomes    Project, dbSNP等已知位点的VCF文件获取已知indels的位置区间，输出Target    intervals    </p>
<p><img src="https://i.imgur.com/Bzem6S3.png" alt=""></p>
<p>（2）通过IndelRealigner进行realignment，输出Realigned BAM</p>
<p>通过–targetIntervals realigner.intervals指定indelrealignment的区间</p>
<p><img src="https://i.imgur.com/murEihF.jpg" alt=""></p>
<p>realignment algorithm 的工作原理：</p>
<p><img src="https://i.imgur.com/Zry1hvH.jpg" alt=""></p>
<h2 id="（4）Base-Recalibration-简称BQSR"><a href="#（4）Base-Recalibration-简称BQSR" class="headerlink" title="（4）Base    Recalibration (简称BQSR)"></a>（4）Base    Recalibration (简称BQSR)</h2><p>纠正机器产生的错误，主要针对测序一产生的碱基质量分数的系统错误，BQSR纠正碱基质量分数（不是碱基本身）</p>
<p><img src="https://i.imgur.com/w7BsDj9.jpg" alt=""></p>
<p>Base Recalibra2on steps/tools</p>
<ul>
<li>Model    the    error    modes    and    compute    adjustments—–》BaseRecalibrator</li>
<li>Apply    recalibra2on adjustments    + write    to    file —–》PrintReads</li>
<li>Make    before/after plots  ——-》AnalyzeCovariates （可选）</li>
</ul>
<p><img src="https://i.imgur.com/lsEMXQk.jpg" alt=""></p>
<p>BaseRecalibrator (1)</p>
<pre><code>java –jar GenomeAnalysisTK.jar \
–T BaseRecalibrator \
–R human.fasta \
–I realigned.bam \
–knownSites dbsnp137.vcf \
–knownSites gold.standard.indels.vcf \
–o recal.table

java –jar GenomeAnalysisTK.jar –T PrintReads \
–R human.fasta \
–I realigned.bam \
–BQSR recal.table \
–o recal.bam
</code></pre><p>BaseRecalibrator (2)(可选)</p>
<p><img src="https://i.imgur.com/ulVOTdY.jpg" alt=""></p>
<pre><code>ava –jar GenomeAnalysisTK.jar –T BaseRecalibrator \
–R human.fasta \
–I realigned.bam \
–knownSites dbsnp137.vcf \
–knownSites gold.standard.indels.vcf \
–BQSR recal.table \
–o after_recal.table

java –jar GenomeAnalysisTK.jar –T AnalyzeCovariates \
–R human.fasta \
–before recal.table \
–after after_recal.table \
–plots recal_plots.pdf
</code></pre><h2 id="（5）Variant-Calling"><a href="#（5）Variant-Calling" class="headerlink" title="（5）Variant Calling"></a>（5）Variant Calling</h2><p>根据变异来源和实验设计不同，选择不同的检测变异方案：</p>
<p><img src="https://i.imgur.com/FkTZcp5.jpg" alt=""></p>
<p>胚系突变检测是最容易的，对于肿瘤体细胞突变主要受肿瘤异质性和正常组织污染的影响，而拷贝数变异本质是计算reads覆盖的丰度</p>
<p>单个基因组是孤立的，通常并不能得出什么变异是致病性变异。一般要求家系或者群体数据，可以发现稀有变异，de novo变异，以及种族背景信息。</p>
<p><img src="https://i.imgur.com/CyxQF14.jpg" alt=""></p>
<p><strong>对于多样本或者家系的突变比较分析，可以使用HaplotypeCaller 的GVCF模式，主要目的GVCF文件输出所有的记录位点，无论其是否是变异。其目的是输出每个样本的每个位点，以便接下来的多样本联合比较分析。</strong></p>
<p><strong>但做多样本的联合分析必须注意，Similar    technical    generation是至关重要的，包括technology,    capture,    read    length,    depth</strong></p>
<p>具体参考：</p>
<ol>
<li><a href="https://gatkforums.broadinstitute.org/gatk/discussion/3893/calling-variants-on-cohorts-of-samples-using-the-haplotypecaller-in-gvcf-mode" target="_blank" rel="external">Calling variants on cohorts of samples using the HaplotypeCaller in GVCF mode</a></li>
<li><a href="https://software.broadinstitute.org/gatk/documentation/article?id=11004" target="_blank" rel="external">GVCF - Genomic Variant Call Format</a></li>
</ol>
<p><img src="https://i.imgur.com/BV54Ah7.jpg" alt=""></p>
<p><img src="https://i.imgur.com/lKHfD2r.jpg" alt=""></p>
<p><img src="https://i.imgur.com/vLF563q.jpg" alt=""></p>
<p><strong>GenotypeGVCFs (i) also has an -allSites    mode and (ii) by default pre-filters    low    QUAL (&lt;30) records</strong></p>
<h3 id="细节"><a href="#细节" class="headerlink" title="细节"></a>细节</h3><h4 id="变异输出模式选择，输出模式总共有3种"><a href="#变异输出模式选择，输出模式总共有3种" class="headerlink" title="变异输出模式选择，输出模式总共有3种"></a>变异输出模式选择，输出模式总共有3种</h4><p>–output_mode / -out_mode (OUTPUT_MODE with default value EMIT_VARIANTS_ONLY)</p>
<ul>
<li>EMIT_VARIANTS_ONLY 只输出variants位点(SNPs和INDELs)</li>
<li>EMIT_ALL_CONFIDENT_SITES 除了variants位点以外与reference一致的可靠位点也输出</li>
<li>EMIT_ALL_SITES 只要是能call的到的位点都输出(应该是除了没覆盖到的位点，其余位点都输出)，这个模式下会得到最多的结果</li>
</ul>
<h4 id="指定区间或者位点（使用-L参数）"><a href="#指定区间或者位点（使用-L参数）" class="headerlink" title="指定区间或者位点（使用-L参数）"></a>指定区间或者位点（使用-L参数）</h4><p>具体可参考：<a href="https://software.broadinstitute.org/gatk/documentation/article.php?id=1319" target="_blank" rel="external">Collected FAQs about interval lists</a></p>
<ul>
<li>对于外显子测序，可以使用.bed文件指定call变异的区间，包含三列（<chr> <start> <end>），以tab键分隔，不需要列名，chr1-chr20,chrX,chrY，并且需要根据坐标排序</end></start></chr></li>
<li><strong>也可以使用vcf文件指定位点call 变异，这对于几个样本的变异位点之间比较很有用</strong></li>
</ul>
<h4 id="Optimizing-and-tuning-the-pipeline"><a href="#Optimizing-and-tuning-the-pipeline" class="headerlink" title="# Optimizing and tuning the pipeline"></a># Optimizing and tuning the pipeline</h4><p>Manage the java heap memory</p>
<p>You can control this by capping the maximum the heap memory can reach using the -Xmx{n} switch. For example, to cap at 8Gb add -Xmx8Gb</p>
<h2 id="（6）Variant-Filtering"><a href="#（6）Variant-Filtering" class="headerlink" title="（6）Variant    Filtering"></a>（6）Variant    Filtering</h2><p>前面提到Base    Recalibration只是纠正碱基质量分数（不是碱基本身），在此Variant Filtering 便是基于Quality    Score    Recalibration的变异过滤。</p>
<p>对于Raw callsets的过滤必须平衡敏感度和准确度</p>
<ul>
<li>Mutation calling算法设计上是宽容的</li>
<li>原始的高敏感度的callests包含大量的假阳性</li>
</ul>
<p>两种过滤方法：</p>
<ul>
<li><p>（1）Hard-filters    using    binary    thresholds    </p>
<ul>
<li>Applicable    to    all    BUT    requires    exper4se    to    define    appropriately    </li>
</ul>
</li>
<li><p>（2）Variant    “recalibration”    using    machine    learning</p>
<ul>
<li>More    powerful    BUT    requires    well-curated    known    resources</li>
</ul>
</li>
</ul>
<p>但是Hard-filtering    is    a    very    blunt    instrument,接下来主要讲述基于variant recalibration的过滤。</p>
<p><strong>基本思想：训练已知的高置信度的位点来确定其他位点正确与否的可能性</strong></p>
<ul>
<li>Assume    annota4ons    tend    to    form    Gaussian    clusters    </li>
<li>Build    a    “Gaussian    mixture    model”    from<br>annotations    of    known    variants    in    our    dataset    </li>
<li>Score    all    variants    by    where    their    annotations    lie<br>relative    to    these    clusters    </li>
<li>Filter    base    on    sensiOvity    to    truth    set    </li>
</ul>
<p><img src="https://i.imgur.com/swA19TJ.jpg" alt=""></p>
<p>那么如何设置VQSLOD阈值呢？即设定X%    of    the    sites in    the    truth<br>set?    </p>
<p><img src="https://i.imgur.com/P8oeuDk.jpg" alt=""></p>
<p><img src="https://i.imgur.com/UKTedbs.png" alt=""></p>
<h3 id="Step-1-VariantRecalibrator"><a href="#Step-1-VariantRecalibrator" class="headerlink" title="Step 1:    VariantRecalibrator"></a>Step 1:    VariantRecalibrator</h3><p>（1）VariantRecalibrator    builds    the    Gaussian    mixture    model    </p>
<ul>
<li>Uses    the    variants    in    the    input    callset    that    overlap    the    training    data    </li>
<li>Uses    the    annotations    in    our    callset,    not    the    resource    callset</li>
</ul>
<p><img src="https://i.imgur.com/LouIHI6.jpg" alt=""></p>
<p><img src="https://i.imgur.com/tCv4gpU.jpg" alt=""></p>
<pre><code>java –jar GenomeAnalysisTK.jar –T VariantRecalibrator \
–R human.fasta \
–input raw.SNPs.vcf \
–resource: {see next slide} \
–an DP –an QD –an FS –an MQRankSum {…} \
–mode SNP \
–recalFile raw.SNPs.recal \
–tranchesFile raw.SNPs.tranches \
–rscriptFile recal.plots.R
</code></pre><p>输出：recal    file,    tranches,    plots</p>
<ul>
<li>Recalibration    plots    show    aspects    of    the    model</li>
<li>Tranche    plots    show    estimated    TP    vs    FP    tradeoff<br>(Default    target    Ti/Tv    is    for    WGS    and    must    be    adapted    for    exomes)</li>
<li>Tranches    :    slices    of    sensitivity    threshold    values</li>
</ul>
<h3 id="Step-2-ApplyRecalibration"><a href="#Step-2-ApplyRecalibration" class="headerlink" title="Step 2: ApplyRecalibration"></a>Step 2: ApplyRecalibration</h3><p>ApplyRecalibration    applies    the    filtering    threshold</p>
<ul>
<li>Executes    the    desired    sensitivity    /    specificity    tradeoff<br>by    applying    filters    to    the    input    callset    (no    new    calculations)    </li>
<li><p>Creates    a    new,    filtered,    analysis-worthy    VCF    file.    </p>
<p>  java –jar GenomeAnalysisTK.jar –T ApplyRecalibration \<br>  –R human.fasta \<br>  –input raw.vcf \<br>  –mode SNP \<br>  –recalFile raw.SNPs.recal \<br>  –tranchesFile raw.SNPs.tranches \<br>  –o recal.SNPs.vcf \<br>  –ts_filter_level 99.0</p>
</li>
</ul>
<p>输出变异每个突变将被注释上VQSLOD score</p>
<h3 id="对于外显子数据跑VQSR的一些小建议"><a href="#对于外显子数据跑VQSR的一些小建议" class="headerlink" title="对于外显子数据跑VQSR的一些小建议"></a>对于外显子数据跑VQSR的一些小建议</h3><ul>
<li>Smaller    number    of    variants    per    sample    compared    to    WGS<br>  -&gt;    typically    insufficient    to    build    a    robust    recalibraOon    model        <pre><code>if    running    on    only    a    few    samples    
</code></pre></li>
<li>Analyze    samples    jointly    in    cohorts    of    at    least    <strong>30    samples</strong></li>
<li>If    necessary,    add    exomes    from    1000G    Project    or    comparable</li>
</ul>
<p><img src="https://i.imgur.com/EoGdjYO.jpg" alt=""></p>
<p><strong>什么时候不推荐使用VQSR进行过滤呢？</strong></p>
<ol>
<li>Non-human    organisms    where    known    resources    are    unavailable    or<br>insufficiently    curated    </li>
<li>RNAseq    data    à    see    RNAseq-specific    filtering</li>
<li>Cohort    is    too    small(&lt;30)    and    no    other    samples    are    available    for<br>“padding”    the    cohort    </li>
</ol>
<p>—–&gt;  Use    manual    filtering    recommendations    instead</p>
<p>具体参考：<a href="https://software.broadinstitute.org/gatk/documentation/article.php?id=39" target="_blank" rel="external">Variant Quality Score Recalibration (VQSR)</a></p>
<p><a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.php" target="_blank" rel="external">VariantRecalibrator</a></p>
<h3 id="（7）-Genotype-Refinement"><a href="#（7）-Genotype-Refinement" class="headerlink" title="（7） Genotype Refinement"></a>（7） Genotype Refinement</h3><p>Using    additional    data    to    improve    genotype    calls    and    likelihoods</p>
<h4 id="基因型校正的重要性："><a href="#基因型校正的重要性：" class="headerlink" title="基因型校正的重要性："></a>基因型校正的重要性：</h4><ol>
<li><p>临床遗传医生需要给病人提供基因型</p>
<ul>
<li>是否病人具有LOF mutation的两个拷贝</li>
<li>患病孩子的父母是否可能生出患该病更严重的孩子</li>
</ul>
</li>
<li><p>群体遗传学家需要基因型数据用于关联研究</p>
<ul>
<li>一个allele的多少拷贝能影响表型</li>
</ul>
</li>
</ol>
<h4 id="用于基因型校正的数据："><a href="#用于基因型校正的数据：" class="headerlink" title="用于基因型校正的数据："></a>用于基因型校正的数据：</h4><ol>
<li>使用高质量的数据（例如1000G）作为先验者</li>
<li>使用家系数据（若可以得到）</li>
<li>计算后验者的基因型可能性</li>
</ol>
<p><img src="https://i.imgur.com/PuPNxd8.jpg" alt=""></p>
<p>基本方法：Bayes’s    Rule</p>
<p><img src="https://i.imgur.com/3Op72W9.jpg" alt=""></p>
<h4 id="基因型校正的工作流程"><a href="#基因型校正的工作流程" class="headerlink" title="基因型校正的工作流程"></a>基因型校正的工作流程</h4><p><img src="https://i.imgur.com/tQP2gCd.jpg" alt=""></p>
<p><img src="https://i.imgur.com/kSzUwtQ.jpg" alt=""></p>
<p>CalculateGenotypePosteriors：</p>
<pre><code>java -jar GenomeAnalysisTK.jar \
-T CalculateGenotypePosteriors \
-R reference.fasta \
-V input.vcf \
-ped family.ped \
-supporting population.vcf \
-o output.vcf
</code></pre><p>Filter    low    confidence    GQs:</p>
<pre><code>java -jar GenomeAnalysisTK.jar \
-T VariantFiltration \
-R reference.fasta \
-V input.vcf \
--filterExpression “GQ&lt;20” \
--filterName “lowGQ” \
-o output.vcf
</code></pre><p>Recommended    threshold is GQ=20    –  GQ 20     is Phred-scaled    99%    confidence</p>
<p>VariantAnnotator:</p>
<pre><code>java -jar GenomeAnalysisTK.jar \
-T VariantAnnotator \
-R reference.fasta \
-V input.vcf \
-A PossibleDeNovo \
-o output.vcf
</code></pre><p><img src="https://i.imgur.com/4gy1NSD.jpg" alt=""></p>
<p>De Novo mutations的特点：</p>
<p><img src="https://i.imgur.com/MjvZPu0.jpg" alt=""></p>
<p>具体参考：</p>
<p>（1）<a href="https://software.broadinstitute.org/gatk/documentation/article.php?id=4723" target="_blank" rel="external">Genotype Refinement workflow</a></p>
<p>（2）<a href="https://software.broadinstitute.org/gatk/documentation/article.php?id=4727" target="_blank" rel="external">(howto) Run the genotype refinement workflow</a></p>
<p>（3）<a href="https://software.broadinstitute.org/gatk/documentation/article.php?id=4726" target="_blank" rel="external">Genotype Refinement workflow: mathematical details</a></p>
<h3 id="（8）-Variant-Evaluation"><a href="#（8）-Variant-Evaluation" class="headerlink" title="（8） Variant Evaluation"></a>（8） Variant Evaluation</h3><p>基本思想：真实变异是具有代表性的，并且是可比较的</p>
<p>变异代表性的影响因素：</p>
<ul>
<li>种群（Ethnicity）:Older     popula+ons    tend    to    display    more    heterogeneity</li>
</ul>
<p>可以作为真实variant sets的数据来源：</p>
<ul>
<li>Sanger sequencing</li>
<li>GeneChip/Microarrays</li>
</ul>
<p>Commonly used truth    sets</p>
<p><img src="https://i.imgur.com/WIRa8Pq.jpg" alt=""></p>
<h4 id="Recommended-metrics-for-callset-evalua-on"><a href="#Recommended-metrics-for-callset-evalua-on" class="headerlink" title="Recommended metrics for    callset    evalua+on"></a>Recommended metrics for    callset    evalua+on</h4><p><img src="https://i.imgur.com/p9i0C4X.jpg" alt=""></p>
<ol>
<li>Indels    &amp;    SNPs的数量，对于单个样本，WGS大约能检测到4.4M个;WES大约能检测到41k个。</li>
<li>如果变异是随机的，那么期望的TiTv Ratio应该为0.5，对于人类WGS测序，TiTv Ratio （Transitions/Transversions）大约为2.0-2.1；对于WES测序，TiTv Ratio大约为3.0-3.3,低的TiTv Ratio暗示可能存在高的假阳性。</li>
<li>Indel Ration表示insertions和deletions的比值，Common的情况Indel Ratio为1； Rare情况下为0.2-0.5。</li>
<li>Genotype    Concordance指标则认为最准确的truth set 是对相同的样本进行genotyping chip，将自己的结果与truth相比较，没有匹配的则认为是假阳性的结果。</li>
</ol>
<p><img src="https://i.imgur.com/oDZl1f7.jpg" alt=""></p>
<p><img src="https://i.imgur.com/p69fkwu.jpg" alt=""></p>
<p>具体参考：<a href="https://software.broadinstitute.org/gatk/documentation/article.php?id=6308" target="_blank" rel="external">Evaluating the quality of a variant callset</a></p>
<h3 id="补充"><a href="#补充" class="headerlink" title="补充"></a>补充</h3><p>生物信息中，一旦涉及到 fastq 数据的处理，通常都要消耗大量内存、CPU 和时间。所以，在正式运行任务之前，建议先从整体数据中抽取一部分小数据进行测试。如何从 fastq 文件中随机抽取数据。使用的工具是 seqtk。</p>
<p>安装：conda install seqtk</p>
<p>使用：</p>
<p>Usage:   seqtk sample [-2] [-s seed=11] <in.fq> <frac>|<number></number></frac></in.fq></p>
<p>Options: -s INT       RNG seed [11]<br>         -2           2-pass mode: twice as slow but with much reduced memory</p>
<p>[]中的是可选参数，&lt;&gt; 中的是必需参数。<br>[-2] 内存较小的服务器上运行时，设置此参数。<br>[-s] 随机数的种子。如果是 Pair-end 数据，需要保证 read1 和 read2 的种子一致，才能抽到相同的raeds。默认是 11。<br>[in.fq] 输入文件</p>
<p><frac|number>可以输入要抽取的比例或 reads 条数。</frac|number></p>
<p>seqtk sample -s100 read1.fastq.gz 10000 | gzip - &gt; sub1.fastq.gz<br>seqtk sample -s100 read2.fastq.gz 10000 | gzip - &gt; sub2.fastq.gz</p>
<h1 id="参考"><a href="#参考" class="headerlink" title="参考"></a>参考</h1><p>（1）<a href="https://mp.weixin.qq.com/s/awdjoXRYobrQAbXmAp3C0g" target="_blank" rel="external">从零开始完整学习全基因组测序数据分析：第4节 构建WGS主流程</a></p>
<p>（2）<a href="http://www.cureffi.org/2012/09/19/exome-sequencing-pipeline-using-gatk/" target="_blank" rel="external">Exome sequencing pipeline using GATK</a></p>

      
    </div>
    
    
    

    

    

    
	<div>
	
		<div>
    
        <div style="text-align:center;color: #ccc;font-size:14px;">-------------本文结束<i class="fa fa-paw"></i>感谢您的阅读-------------</div>
    
</div>
	
	</div>
    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="https:/AnJingwd.github.io/tags/NGS/" rel="tag"><i class="fa fa-tag"></i> NGS</a>
          
        </div>
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="https:/AnJingwd.github.io/2017/12/26/GATK3.8分析环境搭建/" rel="next" title="GATK3.8分析环境搭建">
                <i class="fa fa-chevron-left"></i> GATK3.8分析环境搭建
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="https:/AnJingwd.github.io/2018/01/10/RNA-seq质控/" rel="prev" title="RNA-seq质控">
                RNA-seq质控 <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
        <!-- JiaThis Button BEGIN -->
<div class="jiathis_style">
  <a class="jiathis_button_tsina"></a>
  <a class="jiathis_button_tqq"></a>
  <a class="jiathis_button_weixin"></a>
  <a class="jiathis_button_cqq"></a>
  <a class="jiathis_button_douban"></a>
  <a class="jiathis_button_renren"></a>
  <a class="jiathis_button_qzone"></a>
  <a class="jiathis_button_kaixin001"></a>
  <a class="jiathis_button_copy"></a>
  <a href="http://www.jiathis.com/share" class="jiathis jiathis_txt jiathis_separator jtico jtico_jiathis" target="_blank"></a>
  <a class="jiathis_counter_style"></a>
</div>
<script type="text/javascript" >
  var jiathis_config={
    hideMore:false
  }
</script>
<script type="text/javascript" src="http://v3.jiathis.com/code/jia.js" charset="utf-8"></script>
<!-- JiaThis Button END -->

      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap" >
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image"
               src="https:/AnJingwd.github.io/images/avatar.png"
               alt="AJ's blog" />
          <p class="site-author-name" itemprop="name">AJ's blog</p>
           
              <p class="site-description motion-element" itemprop="description"></p>
           
        </div>
        <nav class="site-state motion-element">

          
            <div class="site-state-item site-state-posts">
              <a href="https:/AnJingwd.github.io/archives">
                <span class="site-state-item-count">57</span>
                <span class="site-state-item-name">日志</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-categories">
              <a href="https:/AnJingwd.github.io/categories/index.html">
                <span class="site-state-item-count">16</span>
                <span class="site-state-item-name">分类</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-tags">
              <a href="https:/AnJingwd.github.io/tags/index.html">
                <span class="site-state-item-count">16</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        
          <div class="feed-link motion-element">
            <a href="https:/AnJingwd.github.io/atom.xml" rel="alternate">
              <i class="fa fa-rss"></i>
              RSS
            </a>
			 <a title="收藏到书签，偶尔High一下^_^" rel="alternate" class="mw-harlem_shake_slow wobble shake" href='javascript:(function() {
    function c() {
        var e = document.createElement("link");
        e.setAttribute("type", "text/css");
        e.setAttribute("rel", "stylesheet");
        e.setAttribute("href", f);
        e.setAttribute("class", l);
        document.body.appendChild(e)
    }
 
    function h() {
        var e = document.getElementsByClassName(l);
        for (var t = 0; t < e.length; t++) {
            document.body.removeChild(e[t])
        }
    }
 
    function p() {
        var e = document.createElement("div");
        e.setAttribute("class", a);
        document.body.appendChild(e);
        setTimeout(function() {
            document.body.removeChild(e)
        }, 100)
    }
 
    function d(e) {
        return {
            height : e.offsetHeight,
            width : e.offsetWidth
        }
    }
 
    function v(i) {
        var s = d(i);
        return s.height > e && s.height < n && s.width > t && s.width < r
    }
 
    function m(e) {
        var t = e;
        var n = 0;
        while (!!t) {
            n += t.offsetTop;
            t = t.offsetParent
        }
        return n
    }
 
    function g() {
        var e = document.documentElement;
        if (!!window.innerWidth) {
            return window.innerHeight
        } else if (e && !isNaN(e.clientHeight)) {
            return e.clientHeight
        }
        return 0
    }
 
    function y() {
        if (window.pageYOffset) {
            return window.pageYOffset
        }
        return Math.max(document.documentElement.scrollTop, document.body.scrollTop)
    }
 
    function E(e) {
        var t = m(e);
        return t >= w && t <= b + w
    }
 
    function S() {
        var e = document.createElement("audio");
        e.setAttribute("class", l);
        e.src = i;
        e.loop = false;
        e.addEventListener("canplay", function() {
            setTimeout(function() {
                x(k)
            }, 500);
            setTimeout(function() {
                N();
                p();
                for (var e = 0; e < O.length; e++) {
                    T(O[e])
                }
            }, 15500)
        }, true);
        e.addEventListener("ended", function() {
            N();
            h()
        }, true);
        e.innerHTML = " <p>If you are reading this, it is because your browser does not support the audio element. We recommend that you get a new browser.</p> <p>";
        document.body.appendChild(e);
        e.play()
    }
 
    function x(e) {
        e.className += " " + s + " " + o
    }
 
    function T(e) {
        e.className += " " + s + " " + u[Math.floor(Math.random() * u.length)]
    }
 
    function N() {
        var e = document.getElementsByClassName(s);
        var t = new RegExp("\\b" + s + "\\b");
        for (var n = 0; n < e.length; ) {
            e[n].className = e[n].className.replace(t, "")
        }
    }
 
    var e = 30;
    var t = 30;
    var n = 350;
    var r = 350;
    var i = "//s3.amazonaws.com/moovweb-marketing/playground/harlem-shake.mp3";
    var s = "mw-harlem_shake_me";
    var o = "im_first";
    var u = ["im_drunk", "im_baked", "im_trippin", "im_blown"];
    var a = "mw-strobe_light";
    var f = "//s3.amazonaws.com/moovweb-marketing/playground/harlem-shake-style.css";
    var l = "mw_added_css";
    var b = g();
    var w = y();
    var C = document.getElementsByTagName("*");
    var k = null;
    for (var L = 0; L < C.length; L++) {
        var A = C[L];
        if (v(A)) {
            if (E(A)) {
                k = A;
                break
            }
        }
    }
    if (A === null) {
        console.warn("Could not find a node of the right size. Please try a different page.");
        return
    }
    c();
    S();
    var O = [];
    for (var L = 0; L < C.length; L++) {
        var A = C[L];
        if (v(A)) {
            O.push(A)
        }
    }
})()'><i class="fa fa-music"></i> High</a>
          </div>
        

        <div class="links-of-author motion-element">
          
        </div>

        
        

        
        
          <div class="links-of-blogroll motion-element links-of-blogroll-inline">
            <div class="links-of-blogroll-title">
              <i class="fa  fa-fw fa-globe"></i>
              推荐阅读
            </div>
            <ul class="links-of-blogroll-list">
              
                <li class="links-of-blogroll-item">
                  <a href="http://blog.sina.com.cn/u/1253988497" title="种田农民的博客" target="_blank">种田农民的博客</a>
                </li>
              
                <li class="links-of-blogroll-item">
                  <a href="https://www.biostars.org/t/Tools/" title="Biostars" target="_blank">Biostars</a>
                </li>
              
                <li class="links-of-blogroll-item">
                  <a href="https://stackoverflow.com/" title="Stack Overflow" target="_blank">Stack Overflow</a>
                </li>
              
                <li class="links-of-blogroll-item">
                  <a href="https://segmentfault.com/" title="SegmentFault" target="_blank">SegmentFault</a>
                </li>
              
                <li class="links-of-blogroll-item">
                  <a href="http://www.alloyteam.com/nav/" title="Web前端导航" target="_blank">Web前端导航</a>
                </li>
              
                <li class="links-of-blogroll-item">
                  <a href="http://www.36zhen.com/t?id=3448" title="前端书籍资料" target="_blank">前端书籍资料</a>
                </li>
              
            </ul>
          </div>
        
		
<div id="music163player">		
	<iframe frameborder="no" border="0" 
	marginwidth="0" marginheight="0" width=330 height=86 
src="//music.163.com/outchain/player?type=2&id=471682981&auto=1&height=66">
	</iframe>
</div>	
        

      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#突变发现背景介绍"><span class="nav-number">1.</span> <span class="nav-text">突变发现背景介绍</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#不同的突变类型"><span class="nav-number">1.1.</span> <span class="nav-text">不同的突变类型</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#实验策略"><span class="nav-number">1.2.</span> <span class="nav-text">实验策略</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Raw-sequence的质控"><span class="nav-number">1.3.</span> <span class="nav-text">Raw sequence的质控</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#胚系突变snp-amp-indel分析流程"><span class="nav-number">2.</span> <span class="nav-text">胚系突变snp&indel分析流程</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#（1）Map"><span class="nav-number">2.1.</span> <span class="nav-text">（1）Map</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#细节："><span class="nav-number">2.1.1.</span> <span class="nav-text">细节：</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#（2）Mark-duplicates"><span class="nav-number">2.2.</span> <span class="nav-text">（2）Mark    duplicates</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#（3）Local-realignment"><span class="nav-number">2.3.</span> <span class="nav-text">（3）Local realignment</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#（4）Base-Recalibration-简称BQSR"><span class="nav-number">2.4.</span> <span class="nav-text">（4）Base    Recalibration (简称BQSR)</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#（5）Variant-Calling"><span class="nav-number">2.5.</span> <span class="nav-text">（5）Variant Calling</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#细节"><span class="nav-number">2.5.1.</span> <span class="nav-text">细节</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#变异输出模式选择，输出模式总共有3种"><span class="nav-number">2.5.1.1.</span> <span class="nav-text">变异输出模式选择，输出模式总共有3种</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#指定区间或者位点（使用-L参数）"><span class="nav-number">2.5.1.2.</span> <span class="nav-text">指定区间或者位点（使用-L参数）</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Optimizing-and-tuning-the-pipeline"><span class="nav-number">2.5.1.3.</span> <span class="nav-text"># Optimizing and tuning the pipeline</span></a></li></ol></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#（6）Variant-Filtering"><span class="nav-number">2.6.</span> <span class="nav-text">（6）Variant    Filtering</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Step-1-VariantRecalibrator"><span class="nav-number">2.6.1.</span> <span class="nav-text">Step 1:    VariantRecalibrator</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Step-2-ApplyRecalibration"><span class="nav-number">2.6.2.</span> <span class="nav-text">Step 2: ApplyRecalibration</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#对于外显子数据跑VQSR的一些小建议"><span class="nav-number">2.6.3.</span> <span class="nav-text">对于外显子数据跑VQSR的一些小建议</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#（7）-Genotype-Refinement"><span class="nav-number">2.6.4.</span> <span class="nav-text">（7） Genotype Refinement</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#基因型校正的重要性："><span class="nav-number">2.6.4.1.</span> <span class="nav-text">基因型校正的重要性：</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#用于基因型校正的数据："><span class="nav-number">2.6.4.2.</span> <span class="nav-text">用于基因型校正的数据：</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#基因型校正的工作流程"><span class="nav-number">2.6.4.3.</span> <span class="nav-text">基因型校正的工作流程</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#（8）-Variant-Evaluation"><span class="nav-number">2.6.5.</span> <span class="nav-text">（8） Variant Evaluation</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Recommended-metrics-for-callset-evalua-on"><span class="nav-number">2.6.5.1.</span> <span class="nav-text">Recommended metrics for    callset    evalua+on</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#补充"><span class="nav-number">2.6.6.</span> <span class="nav-text">补充</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#参考"><span class="nav-number">3.</span> <span class="nav-text">参考</span></a></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <script async src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>

<div class="copyright" >
  
  &copy;  2017 - 
  <span itemprop="copyrightYear">2018</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">AJ's blog</span>
</div>


<!--<div class="powered-by">
  由 <a class="theme-link" href="https://hexo.io">Hexo</a> 强力驱动
</div>

<div class="theme-info">
  主题 -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Pisces
  </a>
</div>
-->


<div class="theme-info">
  <div class="powered-by"></div>
  <span class="post-count">博客全站共75.9k字</span>
</div>

<div class="powered-by">
<i class="fa fa-user-md"></i><span id="busuanzi_container_site_uv">
  本站访客数:<span id="busuanzi_value_site_uv"></span>
</span>
</div>

        

        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  






  
  







  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/three/three.min.js"></script>

  
  <script type="text/javascript" src="https:/AnJingwd.github.io/lib/three/canvas_lines.min.js"></script>


  


  <script type="text/javascript" src="https:/AnJingwd.github.io/js/src/utils.js?v=5.1.2"></script>

  <script type="text/javascript" src="https:/AnJingwd.github.io/js/src/motion.js?v=5.1.2"></script>



  
  


  <script type="text/javascript" src="https:/AnJingwd.github.io/js/src/affix.js?v=5.1.2"></script>

  <script type="text/javascript" src="https:/AnJingwd.github.io/js/src/schemes/pisces.js?v=5.1.2"></script>



  
  <script type="text/javascript" src="https:/AnJingwd.github.io/js/src/scrollspy.js?v=5.1.2"></script>
<script type="text/javascript" src="https:/AnJingwd.github.io/js/src/post-details.js?v=5.1.2"></script>



  


  <script type="text/javascript" src="https:/AnJingwd.github.io/js/src/bootstrap.js?v=5.1.2"></script>



  


  




	





  





  






  





  

  

  

  

  

  


</body>
</html>

<!-- 页面点击小红心 -->
<script type="text/javascript" src="/js/src/love.js"></script>
